src/MetalPerformanceShaders/MPSKernel.cs

```diff
@@ -58,6 +58,15 @@ public static bool Supports (IMTLDevice device)
 	}
 
 #if !COREBUILD
+	public partial class MPSImage {
+
+		[DllImport (Constants.MetalPerformanceShadersLibrary)]
+		static extern MPSImageType MPSGetImageType (IntPtr image);
+
+		[iOS (13,0), TV (12,0), Mac (10,15)][Introduced (PlatformName.MacCatalyst, 13, 0)]
+		public MPSImageType ImageType => MPSGetImageType (Handle);
```

Comment:
mmm this reminds me a little of https://github.com/xamarin/xamarin-macios/issues/6530 (Which I still need to fix heh).
Will the property it be safe for the debugger?
what happens if Handle is invalid (`IntPtr.Zero`)?

Reply:
Fair :-) 

Yes it's safe because, ImageType is just an integer.

The argument is marked nonnull so I assume bad things if null is passed. FWIW, I don't know if any NSObject properties are protected from Zero Handles.

I think this is fine, but I also don't think this function will be used a lot in real code, so I won't push the property.